ext/menoh_native/menoh_ruby.c

```diff
@@ -68,6 +69,9 @@ static menohModel *getModel(VALUE self) {
 
 static void wrap_model_free(menohModel *p) {
   if (p) {
+    if (p->dims) {
+      ruby_xfree(p->dims);
+    }
```

Comment:
You don't need to check if `p->dims` is NULL, because if the argument is NULL `ruby_xfree` do nothing.

Reply:
I fixed it and make it consistent to use for all `ruby_xfree` calls.